hedera-node/hedera-mono-service/src/main/java/com/hedera/node/app/service/mono/state/logic/SigsAndPayerKeyScreen.java

```diff
@@ -62,13 +106,70 @@ public ResponseCodeEnum applyTo(SwirldsTxnAccessor accessor) {
             speedometers.cycleSyncVerifications();
         }
 
+        final var sigMeta = accessor.getSigMeta();
+        if (sigMeta != null) {
+            sigMeta.replacePayerHollowKeyIfNeeded(accessor.getSigMap());
+        }
+
         if (hasActivePayerSig(accessor)) {
             txnCtx.payerSigIsKnownActive();
+
+            if (sigMeta != null && sigMeta.hasReplacedHollowKey()) {
+                accounts.get()
+                        .getForModify(EntityNum.fromAccountId(txnCtx.activePayer()))
+                        .setAccountKey(sigMeta.payerKey());
+                trackHollowAccountCompletion(txnCtx.activePayer(), sigMeta.payerKey());
+            }
+        }
+
+        final var ethTxExpansion = spanMapAccessor.getEthTxExpansion(accessor);
+        if (properties.isLazyCreationEnabled()
+                && ethTxExpansion != null
+                && ethTxExpansion.result().equals(OK)) {
+            final var ethTxSigs = spanMapAccessor.getEthTxSigsMeta(accessor);
+            final var callerNum = aliasManager.lookupIdBy(wrapUnsafely(ethTxSigs.address()));
+            if (callerNum != EntityNum.MISSING_NUM) {
+                var account = accounts.get().getForModify(callerNum);
```

Comment:
`getForModify()` is significantly more expensive than `get()`; so I think we should use `get()` here and then only `getForModify` inside the `if`.

Reply:
Fixed.